From 5e0a2d34f1c03a7d6a1c8591896a21e122d90d6b Mon Sep 17 00:00:00 2001
From: Simon Pichugin <spichugi@redhat.com>
Date: Thu, 23 Jul 2020 23:45:18 +0200
Subject: [PATCH] Issue 51086 - Fix instance name length for interactive
 install

Description: Instance name lenght is not properly validated
during interactive install. Add a check during a user input.

https://pagure.io/389-ds-base/issue/51086

Reviewed by: mreynolds (Thanks!)
---
 src/lib389/lib389/instance/setup.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/lib389/lib389/instance/setup.py b/src/lib389/lib389/instance/setup.py
index f5fc5495d..45c7dfdd4 100644
--- a/src/lib389/lib389/instance/setup.py
+++ b/src/lib389/lib389/instance/setup.py
@@ -308,6 +308,9 @@ class SetupDs(object):
 
             val = input('\nEnter the instance name [{}]: '.format(slapd['instance_name'])).rstrip()
             if val != "":
+                if len(val) > 80:
+                    print("Server identifier should not be longer than 80 symbols")
+                    continue
                 if not all(ord(c) < 128 for c in val):
                     print("Server identifier can not contain non ascii characters")
                     continue
-- 
2.26.2

